tests/kernel/common/prj.conf

```diff
@@ -2,7 +2,6 @@ CONFIG_ZTEST=y
 CONFIG_PRINTK=y
 CONFIG_LOG=y
 CONFIG_POLL=y
-CONFIG_QEMU_TICKLESS_WORKAROUND=y
```

Comment:
Does this remove this workaround for all tests that use it?  If not, it should, and we should remove the whole feature.  It only exists because of timer nondeterminism.

If not, and some things still seem to need it, it implies we have some more work to do somewhere.

Reply:
@andyross In this PR, I remove CONFIG_QEMU_TICKLESS_WORKAROUND in all tests cases, but just three Qemu platforms enabled icount so far, so maybe other Qemu platforms also need it.

Reply:
OK, then if it needs to stay we probably want to force it off at the kconfig level somehow so tests can say "give me the qemu tick workaround if the platform needs it" but the **platform**  layer somewhere can say "this configuration is running under qemu with -icount, don't hack the timer driver".

Reply:
Will change the patch, thanks